### PR TITLE
Stepper: Bulk domain transfer improvements

### PR DESCRIPTION
**Pull request by alshakero**
## Proposed Changes
This does quite a few things to improve the flow dramatically. It addresses feedback from multiple stakeholders. Please infer the changes from the testing steps below.

## Testing Instructions

1. Go to /setup/bulk-domain-transfer

### Better validation
1. The UX should be perfectly smooth with inputting a domain and auth code.
    - No intermediate errors should appear between checks.
2. Please enter an unregistered domain with a random auth code.
    - It should tell you, _This domain does not seem to be registered._
2. Please enter an unlocked domain with a random auth code.
    - It should tell you, _This domain is unlocked but the authentication code seems incorrect._
3. Please enter a locked domain. 
    - It should tell you, _This domain does not seem to be unlocked._

### Secure and smooth persistence
1. Enter a domain and a valid auth code.
2. Go back to intro step.
3. Go forward to domains step.
    - The data should be there.
4. Refresh the page.
    - It should warn you about data loss.
    - Accept, the auth code should be lost, **but not the domain.**
6. Enter a valid domain and auth code.
7. Go to checkout.
8. Click back. 
9. The flow should tell you about the existing domain in the cart and offer going to checkout or starting over. This means the authcodes are stored in the cart, not in local storage.

### Refreshing
1. Enter a valid domain and auth code.
    - Refresh button should be disabled.
2. Enter a valid domain and no auth code.
    - Refresh button should be disabled.
3. Enter a valid domain and an incorrect auth code.
    - Refresh button should refresh when clicked.
3. Enter a valid domain and a correct auth code.
    - Refresh button should be disabled.

### Duplications
1. Enter a domain.
2. Enter another domain with the same name.
3. The second domain should be marked as duplicated.

### CODE REVIEW

**Review by escapemanuele — APPROVED** [review]
Everything worked as advertised! Thanks for improving this 👍

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8067619

Hi @alshakero, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `An unknown error occurred while checking the domain transferability. Please try again or contact support`
- `Please enter a valid authentication code.`

Thank you in advance!

File: client/landing/stepper/declarative-flow/internals/steps-repository/bulk-domain-transfer-domains/use-validation-message.ts [hunk 1]
```diff
@@ -1,39 +1,44 @@
...
-			enabled: Boolean( ! passed && passedLocalValidation && isDomainUnlocked?.unlocked ),
+			enabled: Boolean( ! passed && passedLocalValidation ),
 		}
 	);
 
+	if ( hasDuplicates ) {
+		return {
+			valid: false,
+			loading: false,
+			message: __( 'This domain is a duplicated.' ),
```
Comment (emilyaudela) [inline client/landing/stepper/declarative-flow/internals/steps-repository/bulk-domain-transfer-domains/use-validation-message.ts:38]: Looks like there is a small typo here. It should either be "This domain is duplicated" or "This domain is a duplicate"

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8067619).

**Pull request merged**